src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -698,7 +698,11 @@ public FunctionExpression GetReferenceExpression(ResourceMetadata resource, Synt
                     new JTokenExpression("full"));
             }
 
-            if (resource.IsExistingResource && !context.Settings.EnableSymbolicNames)
+            var shouldIncludeApiVersion =
+                resource.IsExistingResource ||
+                resource is DeclaredResourceMetadata { Symbol.DeclaringResource.Value: IfConditionSyntax };
```

Comment:
Do we have to also handle loops here? `IfConditionSyntax` can be placed inside `ForSyntax`.

Reply:
Good catch! We definitely need to check conditions inside loops. Will push a commit to handle that.

Reply:
Done. Also fixed module output references.